daemon/daemon_unix.go

```diff
@@ -375,6 +390,13 @@ func verifyPlatformContainerSettings(daemon *Daemon, hostConfig *containertypes.
 		warnings = append(warnings, "IPv4 forwarding is disabled. Networking will not work.")
 		logrus.Warnf("IPv4 forwarding is disabled. Networking will not work")
 	}
+	if hostConfig.CgroupParent != "" && daemon.usingSystemd() {
+		// CgroupParent for systemd cgroup should be named as "xxx.slice"
+		length := len(hostConfig.CgroupParent)
+		if length <= 6 || hostConfig.CgroupParent[length-6:] != ".slice" {
```

Comment:
@hqhq But how is that useful to anyone? The **whole point** of being able to set `cgroup-parent` is so you can put a container in a cgroup where you control the resource limitations (like a `slice` in systemd, or just a cgroup that you're managing). **Nobody** "expects" that this will happen (I know, because we've had several cases at SUSE where we saw this and it was completely unexpected and was initially thought to be a Docker bug). If we're going to start restricting `cgroup-parent` we should make sure people understand what's going on.